src/ui/components/CreateIdentifier/components/IdentifierStage1.tsx

```diff
@@ -41,19 +41,15 @@ const IdentifierStage1 = ({
         return textA < textB ? -1 : textA > textB ? 1 : 0;
       });
       setSortedConnections(sortedConnections);
-      setState((prevState: IdentifierStageProps) => ({
-        ...prevState,
-        sortedConnections: sortedConnections,
-      }));
     }
   }, [connectionsCache, setState]);
 
-  const handleSelectConnection = (id: string) => {
+  const handleSelectConnection = (connection: ConnectionShortDetails) => {
     let data = selectedConnections;
-    if (data.find((item) => item === id)) {
-      data = data.filter((item) => item !== id);
+    if (data.find((item) => item === connection)) {
+      data = data.filter((item) => item !== connection);
     } else {
-      data = [...selectedConnections, id];
+      data = [...selectedConnections, connection];
     }
     setSelectedConnections(data);
   };
```

Comment:
Was from a previous PR - but I'm not sure why Redux is updated every time a connection is selected/deselected - and not just once when we click to go to the next screen.